src/status_im/contexts/settings/wallet/saved_addresses/view.cljs

```diff
@@ -71,33 +100,44 @@
 
 (defn view
   []
-  (let [inset-top           (safe-area/get-top)
-        customization-color (rf/sub [:profile/customization-color])
-        saved-addresses     (rf/sub [:wallet/grouped-saved-addresses])]
+  (let [alert-banners-top-margin      (rf/sub [:alert-banners/top-margin])
+        customization-color           (rf/sub [:profile/customization-color])
+        has-saved-addresses?          (rf/sub [:wallet/has-saved-addresses?])
+        [search-text set-search-text] (rn/use-state "")
+        on-clear-input                (rn/use-callback #(set-search-text ""))
+        search-active?                (not (string/blank? search-text))
+        page-top-props                (rn/use-memo
+                                       #(cond-> {:title               (i18n/label :t/saved-addresses)
+                                                 :accessibility-label :saved-addresses-header
+                                                 :title-right         :action
+                                                 :title-right-props   {:icon :i/add
+                                                                       :customization-color
+                                                                       customization-color
+                                                                       :on-press add-address-to-save}
+                                                 :blur?               true}
+
+                                          has-saved-addresses?
+                                          (assoc
+                                           :input       :search
+                                           :input-props {:placeholder         (i18n/label
+                                                                               :t/name-ens-or-address)
+                                                         :value               search-text
+                                                         :on-change-text      set-search-text
```

Comment:
maybe some debounce before call dispatch?

Reply:
Oops! I totally forgot it. NIce catch 👍